code/controllers/subsystem/fire_burning.dm

```diff
@@ -27,9 +27,10 @@ SUBSYSTEM_DEF(fire_burning)
 				return
 			continue
 
-		if(O.resistance_flags & ON_FIRE)
+		if((O.resistance_flags & ON_FIRE) && !(O.resistance_flags & FIRE_PROOF))
 			O.take_damage(20, BURN, "fire", 0)
 		else
+			O.resistance_flags &= ~ON_FIRE //in case we fireproof while burning
 			processing -= O
```

Comment:
actually, this is probably why things sometimes don't have their burning overlays cleared out. do a call for O.extinguished() instead.

Reply:
For context: 
`/obj/proc/extinguish()
	if(resistance_flags & ON_FIRE)
		resistance_flags &= ~ON_FIRE
		cut_overlay(GLOB.fire_overlay, TRUE)
		SSfire_burning.processing -= src
`
Anytime ON_FIRE gets removed, we should already have removed the item from processing.